### DISCUSSION TITLE
Tutorials

### DISCUSSION THREAD

**Discussion by yyoncho** [emacs-lsp/lsp-mode#3165] [General]
- [ ] `Rust Analyzer` - hello world project, using debugger, rust analyzer extensions, refactorings
- [ ] `Python` - setup, virtual envs, debugger, high level comparison of each server
- [ ] `Java` - project setup, debugging, refactoring, unit testing, xml language server for pom.xml
- [ ] `Dart` - debugging, refactoring, unit and widget testing
- [x] `JavaScript` - project setup, eslint, major modes selection, debugging in browser, nodejs, etc.
- [ ] `Scala` - debugger, extensions, ?
- [ ] `Clojure` - using lsp-java to browse java classes. Using dap-mode to debug java code. Using clojure-lsp in standalone and in addition to cider.
- [ ] `OCaml` - setting up `ocaml-lsp` and configuring both handmade and auto-generated (via `dune`) `.merlin` files
- [ ] `Haskell` - setting up `haskell-language-server-wrapper` or `ghcide` with a nontrivial project. Making a custom `hie.yaml` file in cases where `hls`'s automatic project setup fails.
- [ ] `Ruby` - `rbenv`, debugger

**Comment by mpanarin** [comment]
@yyoncho I'd like to start on the tutorial for the python ls. At least for pyls. Do we have any kind of example? Or Format for these?

**Comment by yyoncho** [comment]
I use this as a base https://code.visualstudio.com/docs for javascript tutorials.
This one was written from scratch https://emacs-lsp.github.io/lsp-mode/tutorials/CPP-guide/

**Comment by impo** [comment]
I found this issue while struggling to set up venvs with poetry on Python. A tutorial would be awesome!
